easier68k/core/opcodes/Bcc.py

```diff
@@ -0,0 +1,358 @@
+        raise Exception("Branch Operation did not override Parent's conditional stub")
+        
+    def assemble(self) -> bytearray:
+        """
+        Assembles opcode into Hex
+        :return: Hex for this operation
+        """
+        
+        # yes you actually need to subclass this
+        assert self.cond_code != None
+
+        # hold size
+        extraSize = 2
+        
+        # clear for >8 bit offset
+        lastbyte = '\x00'
+        
+        # if offset is 8 bits than it is inside the opword
+        if self.size == OpSize.BYTE:
+            lastbyte = self.offset.to_bytes()
+            extraSize = 0
+
+        # if offset is 32 bits than the last chunk of opword is FF
+        if self.size == OpSize.LONG:
+            lastbyte = '\xFF'
+            extraSize = 4
+            
+        # first byte = 6+cond_code
+        firstbyte =  6 << 4
+        firstbyte &= self.cond_code.to_bytes()
+        
+        opword = firstbyte << 8
+        opword &= lastbyte
+
+        # convert to bytearray
+        opword = bytearray(opword.to_bytes(2, 'big'))
+
+        # stop here if 8 bit offset
+        if self.size == OpSize.BYTE:
+            return opword
+
+        # add offset
+        opword.extend(self.offset.to_bytes(extraSize, "big"))
+
+        return opword
+
+    
+    def execute(self, simulator: M68K):
+        """
+        Commands the Simulator according to this opcode
+        :param simulator:
+        :return: nothing
+        """
+
+        # check conditional
+        if not self.conditional(simulator):
+            pass
+            
+        simulator.increment_program_counter(self.offset)
+        
+    def __str__(self):
+        """
+        converts command to a string
+        :return: string rep of command
+        """
+
+        return "{s} ${d}".format(
+                s = COND_CODE_TO_OPCODE.get(self.cond_code),
+                d = self.operand or (self.address + 2 + (self.size if self.size > 1 else 0) + self.offset))
+
+    @classmethod
+    def command_matches(cls, command: str) -> bool:
+        """
+        Checks whether a command string is an instance of this command type
+        :param command: The command string to check (e.g. 'MOVE.B', 'LEA', etc.)
+        :return: Whether the string is an instance of this command type
+        """
+        raise Exception("Branch Operation did not override parent class command_matches stub!")
+        #return
+
+    @classmethod
+    def get_word_length(cls, command: str, parameters: str) -> int:
+        """
+        Gets what the end length of this command will be in memory
+        :param command: The text of the command itself (e.g. "LEA", "MOVE.B", etc.)
+        :param parameters: The parameters after the command
+        :return: The length of the bytes in memory in words
+        """
+        
+        # opword
+        length = 2
+        
+        # BIG NOTE: I am explicitly ignoring size codes because there are no defined flippy boys
```

Comment:
Do we have to worry about when the 8-bit displacement is 0 or `0xFF` here, or is that not an issue?

See page 130: https://www.nxp.com/files-static/archives/doc/ref_manual/M68000PRM.pdf

Reply:
To be sure, I compiled the following code in Easy68k and got "60FE" back out of it
```
L    BRA L
```
I will just add a line to the constructor of the branch_code base class that sets displacement to -1 when it is 0.

Reply:
"Note: A branch to the immediately following instruction automatically
uses   the   16-bit   displacement   format   because   the   8-bit
displacement field contains $00 (zero offset)." 

0xFF on the other hand, is an illegal access exception, as you would instruct the sim to pull a word at an odd byte-address, causing the sim to need two read operations to get the word.

Two corner cases which definitely need to be added to my code.